client/signup/main.jsx

```diff
@@ -238,7 +238,15 @@ const Signup = React.createClass( {
 		if ( userIsLoggedIn ) {
 			// deferred in case the user is logged in and the redirect triggers a dispatch
 			defer( function() {
-				page( destination );
+				/**
+				 * The destination URL is not always inside Calypso.
+				 * For these cases make sure that a proper redirect is issued.
+				 */
+				if ( /^https?:\/\// ) {
```

Comment:
This works for me, and I can't think of a way that it'd currently mess up. No SSR to worry about. Might be worth adding a note to the `destination` explanation in the Signup readme, but other than that, :+1:.